### ISSUE TITLE
Injection issue with Micronaut Http Client guide running on docker container

### ISSUE THREAD

**Issue by Reach-NickLavalle** [micronaut-projects/micronaut-core#6176]
### Expected Behavior

Doing a dockerBuildNative of the Micronaut Http Client would allow me to run the client from a container and still return the releases from the /github/releases-lowlevel URL. Forgive me but I'm new to micronaut so if this is an obvious issue or it's expected the guide won't support a docker container image of it then I apologize.

### Actual Behaviour

The URL returns:
`"Internal Server Error: Failed to inject value for parameter [configuration] of class: example.micronaut.GithubLowLevelClient\n\nMessage: No bean of type [example.micronaut.GithubConfiguration] exists. Make sure the bean is not disabled by bean requirements (enable trace logging for 'io.micronaut.context.condition' to check) and if the bean is enabled then ensure the class is declared a bean and annotation processing is enabled (for Java and Kotlin the 'micronaut-inject-java' dependency should be configured as an annotation processor).\nPath Taken: new GithubController(GithubLowLevelClient githubLowLevelClient,GithubApiClient githubApiClient) --> new GithubController([GithubLowLevelClient githubLowLevelClient],GithubApiClient githubApiClient) --> new GithubLowLevelClient(HttpClient httpClient,[GithubConfiguration configuration])"`

### Steps To Reproduce

1. Downloaded the zipped Micronaut Http Client solution from https://guides.micronaut.io/latest/micronaut-http-client-gradle-java.html
2. Confirm URL body response with a `./gradlew assemble` and running the *.jar, the /github/releases-lowlevel URL returns the expected json result body.
3. Added 
```
dockerfileNative {
 instruction "EXPOSE 8080"
}
```
to the build.gradle file to expose the same port on the container.
4. Run `./gradlew dockerBuildNative` to build the container image
5. Run the container with a `docker run -d -p 8080:8080 micronautguide`
6. Attempt an HTTP GET to the localhost:8080/github/releases-lowlevel 

### Environment Information

- Operating System: Windows 10
- Micronaut version: 3.0.1
- openjdk 11.0.12 2021-07-20
- OpenJDK Runtime Environment GraalVM CE 21.2.0 (build 11.0.12+6-jvmci-21.2-b08)
- OpenJDK 64-Bit Server VM GraalVM CE 21.2.0 (build 11.0.12+6-jvmci-21.2-b08, mixed mode, sharing)

### Version

3.0.1

**Comment by Reach-NickLavalle** [comment]
A discovery I have had is that the dockerBuildNative does **not** copy the application.yml or set the environment variables. Updating my build.gradle piece to this allowed the URL to process.
```
dockerfileNative {
 instruction "EXPOSE 8080"
 instruction "COPY --from=graalvm /home/app/resources/application.yml /app/application.yml"
 instruction "ENV MICRONAUT_CONFIG_FILES=/app/application.yml" 
}
```

Is this something that could be added to the dockerBuildNative command? Does that belong on the plugin repo instead of here?

**Comment by ilopmar** [comment]
I'm not able to reproduce the issue with the latest version of the guide (currently Micronaut 3.1.0).

@Reach-NickLavalle please double-check the version of the Micronaut Gradle plugin you are using. There has been a few issues with resources not included in the native image when building inside Docker. All of them should be fixed now. The latest one is

```
id("io.micronaut.application") version "2.0.6"
```

Btw, you don't need to include the `EXPOSE 8080`. For me it everything works out of the box (unzip project, run `./gradlew dockerBuildNative` and start the container with `docker run -it --rm -p 8080:8080 micronautguide:latest`

If it doesn't work with you with the latest version of the guide (that already includes the latest version of the Micronaut Gradle plugin), please file a new issues with all the steps.

**Comment by Reach-NickLavalle** [comment]
Iván updating the plugin version to 2.0.6 resolved my case as well, it was 2.0.4 before. Looks like it must've been something between those versions. Thanks for closing that out this simplifies my workflow.
